sdk/src/Core/Amazon.Runtime/Credentials/FederatedAWSCredentials.cs

```diff
@@ -189,28 +195,37 @@ private CredentialsRefreshState Authenticate(ICredentials userCredential)
                 region = DefaultSTSClientRegion;
             }
 
-            ICoreAmazonSTS coreSTSClient = null;
-            try
+            ICoreAmazonSTS coreSTSClient = GlobalRuntimeDependencyRegistry.Instance.GetInstance<ICoreAmazonSTS>(ServiceClientHelpers.STS_ASSEMBLY_NAME, ServiceClientHelpers.STS_SERVICE_CLASS_NAME,
+                new CreateInstanceContext(new SecurityTokenServiceClientContext(SecurityTokenServiceClientContext.ActionContext.FederatedAWSCredentials, region)));
```

Comment:
Below it is using `AnonymousAWSCredentials`. Should there be `AnonymousAWSCredentials` under *.ActionContext. ?

Reply:
I added a comment on the `ActionContext` enum that for `FederatedAWSCredentials` the user can use `AnonymousAWSCredentials`. I don't think we need to put an actual instance of `AnonymousAWSCredentials` on the context object and would probably cause confusion for the other use causes.